Sources/GDSCommon/UI/UIKit/SharedViews/Patterns/Scanner/ScanningViewController.swift

```diff
@@ -0,0 +1,298 @@
+    
+    private let processingQueue = DispatchQueue(label: "barcodeScannerQueue",
+                                                qos: .userInitiated,
+                                                attributes: [],
+                                                autoreleaseFrequency: .workItem)
+    
+    /// Initialiser for the `Scanning` view controller.
+    /// Requires a single parameter.
+    /// - Parameter scanningController: `ScanningController`
+    /// - Parameter viewModel: `QRScanningViewModel`
+    public init(scanningController: ScanningController,
+                viewModel: QRScanningViewModel) {
+        self.scanningController = scanningController
+        self.viewModel = viewModel
+        
+        captureSession = AVCaptureSession()
+        previewLayer = AVCaptureVideoPreviewLayer(session: captureSession)
+        presenter = DialogView<CheckmarkDialogAccessoryView>(title: viewModel.successMessage,
+                                                             isLoading: false)
+        super.init(nibName: "Scanner", bundle: .module)
+        barcodeRequest = VNDetectBarcodesRequest(completionHandler: detectedBarcode(_:_:))
+    }
+    
+    required public init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+        title = viewModel.title
+    }
+    
+    public override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        makeScannerCaptureView()
+        updateRegionOfInterest()
+        addImageOverlay()
+        previewLayer.frame = cameraView.layer.bounds
+    }
+    
+    public override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        startAnimation()
+    }
+    
+    private func showLoadingDialog(withText text: String? = nil) async {
+        guard let overlayView else { return }
+        await presenter.present(onView: overlayView, shouldLoad: false, title: text)
+    }
+    
+    private func updateRegionOfInterest() {
+        previewLayer.layoutIfNeeded()
+        cameraView.layoutIfNeeded()
+    }
+    
+    public func captureOutput(_ output: AVCaptureOutput,
+                              didOutput sampleBuffer: CMSampleBuffer,
+                              from connection: AVCaptureConnection) {
+        guard isScanning,
+              let pixelBuffer = CMSampleBufferGetImageBuffer(sampleBuffer) else { return }
+        let handler = VNImageRequestHandler(cvPixelBuffer: pixelBuffer, options: [:])
+        do {
+            try handler.perform([barcodeRequest])
+        } catch {
+            preconditionFailure("Error with capturing output")
+        }
+    }
+    
+    func startAnimation() {
+        UIView.animate(withDuration: 1.0) {
+            self.imageView.transform = CGAffineTransform(scaleX: 0.95, y: 0.95)
+        } completion: {(_ finished: Bool) -> Void in
```

Comment:
Do you need the closure signature here `(_ finished: Bool) -> Void in` if you're not using the argument? Maybe you can't get rid of it? 🤷‍♂️